docker/init.sh

```diff
@@ -0,0 +1,45 @@
+#! /bin/bash
+
+set -e
+
+# data directory is required for caching the secret key in a file
+if [ "$DATA_DIR" == "" ]; then
+    echo -n "Error: DATA_DIR environment variable not set!"
+    echo "Are you sure you are running this script in a Docker container?"
+    exit 1
+fi
+
+SECRET_KEY_FILE="$DATA_DIR/secret_key"
+
+if [ ! -f "$SECRET_KEY_FILE" ]; then
```

Comment:
As mentioned before, I don't have too much experience with either Ruby or Rails. This method worked quite well, and the key is cached in a named volume.

This could also obsolete having to set `SECRET_KEY_BASE` in `docker-compose.env`. And it brings up the question: Should the Docker container still autogenerate the key(s) and store them in a volume?

At the moment, there's the following options:

  * Autogenerate the keys (and optionally store them in a volume, otherwise they had to be regenerated on updates). Later, the user could use `docker-compose run bundle exec rails secrets`to regenerate them. The only problem would be how to put the `secrets.yml` in a container, but I guess I could figure that out (a symlink would probably work).
  * Have the user create a `secrets.yml` on the host that is mounted as a volume. This is something I'd really like to avoid, because it requires Ruby and all the deps on the host, too, which eliminates many of Docker's advantages. This would also create a bunch of security issues because a user would have to make sure the file is kept secret.
  * Make the user set `SECRET_KEY_BASE` and `SECRET_KEY` in the `docker-compose.env` file. This might work well, too, but doesn't sound like the best option either.

I would go for the first approach.